include/ck_tile/ops/gemm/kernel/gemm_tile_partitioner.hpp

```diff
@@ -364,4 +365,444 @@ struct GemmSpatiallyLocalTilePartitioner
     index_t N;
 };
 
+/**
+ * @brief Stream-K tile partitioner that dynamically balances work across workgroups
+ *
+ * This partitioner is responsible for mapping workgroups to tiles in the C tensor
+ * for the Stream-K algorithm which decomposes the GEMM problem
+ * into smaller work units and distributes them more evenly across available blocks,
+ * improving load balancing especially for cases where the K dimension is large.
+ */
+template <typename BlockGemmShape,
+          StreamKReductionStrategy ReductionStrategy = ck_tile::StreamKReductionStrategy::Atomic,
+          index_t TileSwizzleSubM                    = 8>
+struct StreamKTilePartitioner
+{
+
+    static constexpr index_t MPerBlock = BlockGemmShape::kM;
+    static constexpr index_t NPerBlock = BlockGemmShape::kN;
+    static constexpr index_t KPerBlock = BlockGemmShape::kK;
+
+    index_t sk_num_blocks;
+    index_t sk_num_big_blocks;
+    index_t dp_start_block_idx;
+    index_t reduction_start_block_idx;
+    index_t k_iters_per_big_block;
+    mdiv2 n_tiles;
+    mdiv k_iters_per_tile;
+    mdiv equiv_tiles_big;    // for reduction
+    mdiv equiv_tiles_little; // for reduction
+
+    CK_TILE_HOST_DEVICE StreamKTilePartitioner() noexcept = delete;
+
+    /**
+     * @brief Construct Stream-K tile partitioner with problem dimensions
+     */
+    CK_TILE_HOST_DEVICE StreamKTilePartitioner(index_t M,
+                                               index_t N,
+                                               index_t K,
+                                               index_t num_cu,
+                                               index_t occupancy,
+                                               index_t sk_blocks = -1) noexcept
```

Comment:
MNK, num_cu, occupancy and sk_blocks can probably be uint32_t. Negative values aren't likely, plus have a bigger range of supported values (especially if MNK get really large)